webservices/legal_docs/advisory_opinions.py

```diff
@@ -60,7 +67,7 @@
 """
 
 STATUTE_CITATION_REGEX = re.compile(
-    r"(?P<title>\d+)\s+U\.?S\.?C\.?\s+§*\s*(?P<section>\d+).*\.?")
+    r"(?P<title>\d+)\s+U\.?S\.?C\.?\s+§*\s*(?P<section>\d+\d+[-0-9a-z]*).*\.?")
```

Comment:
This has the undesired consequence of requiring that section numbers be at least _2_ digits followed by 0 or more digit-or-lowercase-alphabet-or-dash.

e.g. try parsing `52 U.S.C. 3`.

Reply:
Thank you, I've actually spent some time trying to understand REGEX better and I have a long way to go. It looks like the 2nd `\d+` is the problem, so it should be `(?P<title>\d+)\s+U\.?S\.?C\.?\s+§*\s*(?P<section>\d+[-0-9a-z]*).*\.?`. Do you agree?

Reply:
`(?P<section>\d+[-0-9a-z]*)` matches "one or more digits that may _optionally_ be followed by 0 or more digit-or-lowercase-alphabet-or-dash and would also match "123a-4".

`(?P<section>\d+[-a-z]*)` matches "one or more digits that may _optionally_ be followed by 0 or more lowercase-alphabet-or-dash and would match "123a" and "123a-b" but not "123a-4".

Not sure which one is better, but my hunch is the latter.

Reply:
Thanks @vrajmohan! I think we want to catch  2 USC 441a-1 so I'll test the first option.  http://uscode.house.gov/editorialreclassification/t52/Reclassifications_Title_52.html